In [1]:
%load_ext sql

In [2]:
%sql mysql+pymysql://root:password@localhost:3306/md_water_services

In [3]:
%%sql 

SELECT
    *
FROM
    data_dictionary

 * mysql+pymysql://root:***@localhost:3306/md_water_services
49 rows affected.


[('employee', 'assigned_employee_id', 'Unique ID assigned to each employee', 'INT', 'visits'),
 ('employee', 'employee_name', 'Name of the employee', 'VARCHAR(255)', ''),
 ('employee', 'phone_number', 'Contact number of the employee', 'VARCHAR(15)', ''),
 ('employee', 'email', 'Email address of the employee', 'VARCHAR(255)', ''),
 ('employee', 'address', 'Residential address of the employee', 'VARCHAR(255)', ''),
 ('employee', 'town_name', 'Name of the town where the employee resides', 'VARCHAR(255)', ''),
 ('employee', 'province_name', 'Name of the province where the employee resides', 'VARCHAR(255)', ''),
 ('employee', 'position', 'Position or job title of the employee', 'VARCHAR(255)', ''),
 ('visits', 'record_id', 'Unique ID assigned to each visit', 'int', 'water_quality, water_source'),
 ('visits', 'location_id', 'ID of the location visited', 'varchar(255)', 'location'),
 ('visits', 'source_id', 'ID of the water source visited', 'varchar(510)', 'well_pollution'),
 ('visits', 'time_of_record', 'Date and time of the visit', 'datetime', ''),
 ('visits', 'visit_count', 'Number of visits made to this location', 'int', ''),
 ('visits', 'time_in_queue', 'Time spent by people waiting for water in a queue at the location', 'int', ''),
 ('visits', 'assigned_employee_id', 'ID of the employee who visited the location', 'int', 'employee'),
 ('water_quality', 'record_id', 'Unique ID assigned to each record', 'int', ''),
 ('water_quality', 'subjective_quality_score', 'Score representing the subjective quality of the water source', 'int', ''),
 ('water_quality', 'visit_count', 'Number of visits made for data collection', 'int', ''),
 ('water_source', 'source_id', 'Unique ID assigned to each water source', 'VARCHAR(510)', 'visits'),
 ('water_source', 'type_of_water_source', 'Type or category of the water source. Can be : tap_in_home, tap_in_home_broken, well, shared_tap, river', 'VARCHAR(255)', ''),
 ('water_source', 'number_of_people_served', 'Number of people served by this water source', 'INT', ''),
 ('well_pollution', 'source_id', 'ID of the water source being tested for pollution', 'varchar(258)', 'visits'),
 ('well_pollution', 'date', 'Date of the pollution test', 'datetime', ''),
 ('well_pollution', 'description', 'Description of the pollution test', 'varchar(255)', ''),
 ('well_pollution', 'pollutant_ppm', 'Result of the pollution test in parts per million', 'float', ''),
 ('well_pollution', 'biological', 'Biological contamination level', 'float', ''),
 ('well_pollution', 'results', "Result of the pollution test. Can be 'dirty', 'biologically contaminated', or 'clean'", 'varchar(255)', ''),
 ('location', 'location_id', 'Unique ID assigned to each location', 'varchar(255)', 'visits'),
 ('location', 'address', 'Address of the location', 'varchar(255)', ''),
 ('location', 'province_name', 'Name of the province where the location is situated', 'varchar(255)', ''),
 ('location', 'town_name', 'Name of the town where the location is situated', 'varchar(255)', ''),
 ('location', 'location_type', 'Type or category of the location', 'varchar(255)', ''),
 ('global_water_access', 'name', 'The country or area name', 'varchar(255)', ''),
 ('global_water_access', 'region', 'Geographical region', 'varchar(255)', ''),
 ('global_water_access', 'year', 'Year of the data record', 'int', ''),
 ('global_water_access', 'pop_n', 'The national population size estimate in thousands', 'float', ''),
 ('global_water_access', 'pop_u', 'The urban population share estimate in percentage points (%)', 'float', ''),
 ('global_water_access', 'wat_bas_n', 'The estimated national share of people with at least basic service (%)', 'float', ''),
 ('global_water_access', 'wat_lim_n', 'The estimated national share of people with limited service (%)', 'float', ''),
 ('global_water_access', 'wat_unimp_n', 'The estimated national share of people with unimproved service (%)', 'float', ''),
 ('global_water_access', 'wat_sur_n', 'The estimated national share of people with surfa

# 1-Clean our Data

In [15]:
# %%sql

# SELECT
#     CONCAT (
#         LOWER(REPLACE (employee_name, " ", '.')),
#         '@ndogowater.gov'
#     ) AS new_email,
#     email
# FROM
#     employee
# LIMIT
#     5

In [13]:
# %%sql

# update employee set email = CONCAT (LOWER(REPLACE (employee_name, " ", '.')),'@ndogowater.gov')

 * mysql+pymysql://root:***@localhost:3306/md_water_services
56 rows affected.


[]

In [20]:
# %%sql 

# SELECT
#     phone_number,length(phone_number),trim(phone_number) as new_phone_number,length(trim(phone_number))
# FROM
#     employee
# LIMIT
#     5

 * mysql+pymysql://root:***@localhost:3306/md_water_services
5 rows affected.


[('+99637993287 ', 13, '+99637993287', 12),
 ('+99643864786 ', 13, '+99643864786', 12),
 ('+99222599041 ', 13, '+99222599041', 12),
 ('+99945849900 ', 13, '+99945849900', 12),
 ('+99381679640 ', 13, '+99381679640', 12)]

In [21]:
# %%sql

# UPDATE employee
# SET
#     phone_number = TRIM(phone_number)

 * mysql+pymysql://root:***@localhost:3306/md_water_services
56 rows affected.


[]

In [22]:
# %%sql 

# SELECT
#     phone_number,
#     length (phone_number),
#     TRIM(phone_number) AS new_phone_number,
#     length (TRIM(phone_number))
# FROM
#     employee
# LIMIT
#     5

 * mysql+pymysql://root:***@localhost:3306/md_water_services
5 rows affected.


[('+99637993287', 12, '+99637993287', 12),
 ('+99643864786', 12, '+99643864786', 12),
 ('+99222599041', 12, '+99222599041', 12),
 ('+99945849900', 12, '+99945849900', 12),
 ('+99381679640', 12, '+99381679640', 12)]

# 2-Honouring the workers

In [4]:
# %%sql 

# SELECT
#     COUNT(employee_name) AS "Number of Employees",
#     town_name AS "Town"
# FROM
#     employee
# GROUP BY
#     2

In [5]:
%%sql 

SELECT
    COUNT(v.visit_count) AS visit_count,
    v.assigned_employee_id,
    employee_name,
    phone_number,
    email
FROM
    visits v
    JOIN employee e ON e.assigned_employee_id = v.assigned_employee_id
GROUP BY
    2
ORDER BY
    1 DESC
LIMIT
    3

 * mysql+pymysql://root:***@localhost:3306/md_water_services
3 rows affected.


[(3708, 1, 'Bello Azibo', '+99643864786', 'bello.azibo@ndogowater.gov'),
 (3676, 30, 'Pili Zola', '+99822478933', 'pili.zola@ndogowater.gov'),
 (3539, 34, 'Rudo Imani', '+99046972648', 'rudo.imani@ndogowater.gov')]

In [155]:
%%sql 

SELECT
    COUNT(v.visit_count) AS visit_count,
    v.assigned_employee_id,
    employee_name,
    phone_number,
    email
FROM
    visits v
    JOIN employee e ON e.assigned_employee_id = v.assigned_employee_id
GROUP BY
    2
ORDER BY
    1 
LIMIT
    3

 * mysql+pymysql://root:***@localhost:3306/md_water_services
3 rows affected.


[(15, 20, 'Kunto Asha', '+99176320477', 'kunto.asha@ndogowater.gov'),
 (143, 22, 'Lesedi Kofi', '+99611183730', 'lesedi.kofi@ndogowater.gov'),
 (218, 44, 'Wambui Jabali', '+99269319001', 'wambui.jabali@ndogowater.gov')]

# 3-Analysing locations

In [17]:
%%sql

SELECT
    COUNT(location_id),
    town_name
FROM
    location
GROUP BY
    2
ORDER BY
    1 desc

 * mysql+pymysql://root:***@localhost:3306/md_water_services
25 rows affected.


[(23740, 'Rural'),
 (1650, 'Harare'),
 (1090, 'Amina'),
 (1070, 'Lusaka'),
 (990, 'Mrembo'),
 (930, 'Asmara'),
 (930, 'Dahabu'),
 (780, 'Kintampo'),
 (780, 'Ilanga'),
 (770, 'Isiqalo'),
 (750, 'Zuri'),
 (710, 'Amara'),
 (620, 'Majengo'),
 (520, 'Pwani'),
 (510, 'Kofi'),
 (480, 'Serowe'),
 (470, 'Bahari'),
 (460, 'Marang'),
 (400, 'Abidjan'),
 (400, 'Bello'),
 (370, 'Cheche'),
 (350, 'Deka'),
 (310, 'Djenne'),
 (300, 'Zanzibar'),
 (270, 'Yaounde')]

In [16]:
%%sql

SELECT
    COUNT(location_id),
    province_name
FROM
    location
GROUP BY
    2
ORDER BY
    1 desc

 * mysql+pymysql://root:***@localhost:3306/md_water_services
5 rows affected.


[(9510, 'Kilimani'),
 (8940, 'Akatsi'),
 (8220, 'Sokoto'),
 (6950, 'Amanzi'),
 (6030, 'Hawassa')]

In [19]:
%%sql

SELECT
    province_name,
    town_name,
    COUNT(location_id) AS records_per_town
FROM
    location
GROUP BY
    1,
    2
ORDER BY
    1,
    3 desc

 * mysql+pymysql://root:***@localhost:3306/md_water_services
31 rows affected.


[('Akatsi', 'Rural', 6290),
 ('Akatsi', 'Lusaka', 1070),
 ('Akatsi', 'Harare', 800),
 ('Akatsi', 'Kintampo', 780),
 ('Amanzi', 'Rural', 3100),
 ('Amanzi', 'Asmara', 930),
 ('Amanzi', 'Dahabu', 930),
 ('Amanzi', 'Amina', 670),
 ('Amanzi', 'Pwani', 520),
 ('Amanzi', 'Abidjan', 400),
 ('Amanzi', 'Bello', 400),
 ('Hawassa', 'Rural', 3900),
 ('Hawassa', 'Serowe', 480),
 ('Hawassa', 'Amina', 420),
 ('Hawassa', 'Deka', 350),
 ('Hawassa', 'Djenne', 310),
 ('Hawassa', 'Zanzibar', 300),
 ('Hawassa', 'Yaounde', 270),
 ('Kilimani', 'Rural', 5440),
 ('Kilimani', 'Mrembo', 990),
 ('Kilimani', 'Harare', 850),
 ('Kilimani', 'Isiqalo', 770),
 ('Kilimani', 'Zuri', 750),
 ('Kilimani', 'Amara', 710),
 ('Sokoto', 'Rural', 5010),
 ('Sokoto', 'Ilanga', 780),
 ('Sokoto', 'Majengo', 620),
 ('Sokoto', 'Kofi', 510),
 ('Sokoto', 'Bahari', 470),
 ('Sokoto', 'Marang', 460),
 ('Sokoto', 'Cheche', 370)]

In [32]:
%%sql

SELECT
    COUNT(location_id) AS records_per_town,
    location_type
FROM
    location
GROUP BY
    2

 * mysql+pymysql://root:***@localhost:3306/md_water_services
2 rows affected.


[(15910, 'Urban'), (23740, 'Rural')]

In [34]:
%%sql
SELECT
    (23740 / (15910 + 23740) * 100),
    (15910 / (15910 + 23740) * 100)

 * mysql+pymysql://root:***@localhost:3306/md_water_services
1 rows affected.


[(Decimal('59.8739'), Decimal('40.1261'))]

# 4-Diving into the sources


In [36]:
%%sql

SELECT
    *
FROM
    water_source
LIMIT
    10

 * mysql+pymysql://root:***@localhost:3306/md_water_services
10 rows affected.


[('AkHa00000224', 'tap_in_home', 956),
 ('AkHa00001224', 'tap_in_home_broken', 930),
 ('AkHa00002224', 'tap_in_home_broken', 486),
 ('AkHa00003224', 'well', 364),
 ('AkHa00004224', 'tap_in_home_broken', 942),
 ('AkHa00005224', 'tap_in_home', 736),
 ('AkHa00006224', 'tap_in_home', 882),
 ('AkHa00007224', 'tap_in_home', 554),
 ('AkHa00008224', 'well', 398),
 ('AkHa00009224', 'well', 346)]

1. How many people did we survey in total?

In [52]:
%%sql

SELECT
    sum(number_of_people_served)
FROM
    water_source

 * mysql+pymysql://root:***@localhost:3306/md_water_services
1 rows affected.


[(Decimal('27628140'),)]

2. How many wells, taps and rivers are there?


In [44]:
%%sql

SELECT
    COUNT(type_of_water_source),
    type_of_water_source
FROM
    water_source
GROUP BY
    2

 * mysql+pymysql://root:***@localhost:3306/md_water_services
5 rows affected.


[(7265, 'tap_in_home'),
 (5856, 'tap_in_home_broken'),
 (17383, 'well'),
 (5767, 'shared_tap'),
 (3379, 'river')]

3. How many people share particular types of water sources on average?

In [47]:
%%sql

SELECT
    round(avg(number_of_people_served)),
    type_of_water_source
FROM
    water_source
GROUP BY
    2

 * mysql+pymysql://root:***@localhost:3306/md_water_services
5 rows affected.


[(Decimal('644'), 'tap_in_home'),
 (Decimal('649'), 'tap_in_home_broken'),
 (Decimal('279'), 'well'),
 (Decimal('2071'), 'shared_tap'),
 (Decimal('699'), 'river')]

This means that 1 tap_in_home actually represents 644 ÷ 6 = ± 100 taps.


4. How many people are getting water from each type of source?


In [55]:
%%sql

SELECT
    SUM(number_of_people_served),
    type_of_water_source
FROM
    water_source
GROUP BY
    2
ORDER BY
    1 desc

 * mysql+pymysql://root:***@localhost:3306/md_water_services
5 rows affected.


[(Decimal('11945272'), 'shared_tap'),
 (Decimal('4841724'), 'well'),
 (Decimal('4678880'), 'tap_in_home'),
 (Decimal('3799720'), 'tap_in_home_broken'),
 (Decimal('2362544'), 'river')]

In [56]:
%%sql

SELECT
    round((SUM(number_of_people_served)/27628140)*100),
    type_of_water_source
FROM
    water_source
GROUP BY
    2
ORDER BY
    1 desc

 * mysql+pymysql://root:***@localhost:3306/md_water_services
5 rows affected.


[(Decimal('43'), 'shared_tap'),
 (Decimal('18'), 'well'),
 (Decimal('17'), 'tap_in_home'),
 (Decimal('14'), 'tap_in_home_broken'),
 (Decimal('9'), 'river')]

43% of our people are using shared taps in their communities, and on average, we saw earlier, that 2000 people share one shared_tap.


By adding tap_in_home and tap_in_home_broken together, we see that 31% of people have water infrastructure installed in their homes, but 45%
(14/31) of these taps are not working! This isn't the tap itself that is broken, but rather the infrastructure like treatment plants, reservoirs, pipes, and
pumps that serve these homes that are broken.

18% of people are using wells. But only 4916 out of 17383 are clean = 28% (from last week).

# 5-Start of a solution


In [78]:
%%sql

SELECT
    SUM(number_of_people_served) as population_served,
    type_of_water_source,
    RANK() OVER (
        ORDER BY
            SUM(number_of_people_served) DESC
    ) AS rank_by_population
FROM
    water_source
GROUP BY
    2


 * mysql+pymysql://root:***@localhost:3306/md_water_services
5 rows affected.


[(Decimal('11945272'), 'shared_tap', 1),
 (Decimal('4841724'), 'well', 2),
 (Decimal('4678880'), 'tap_in_home', 3),
 (Decimal('3799720'), 'tap_in_home_broken', 4),
 (Decimal('2362544'), 'river', 5)]

In [4]:
%%sql

SELECT
    source_id,
    type_of_water_source,
    number_of_people_served,
    ROW_NUMBER() OVER (
        PARTITION BY
            type_of_water_source
        ORDER BY
            number_of_people_served DESC
    ) AS priority_rank
FROM
    water_source
LIMIT
    5

 * mysql+pymysql://root:***@localhost:3306/md_water_services
5 rows affected.


[('SoRu36791224', 'river', 998, 1),
 ('SoRu36238224', 'river', 998, 2),
 ('SoRu36880224', 'river', 998, 3),
 ('SoRu37756224', 'river', 998, 4),
 ('SoRu38142224', 'river', 998, 5)]

# 6-Analysing queues

In [91]:
%%sql

SELECT
    *
FROM
    visits

limit 10

 * mysql+pymysql://root:***@localhost:3306/md_water_services
10 rows affected.


[(0, 'SoIl32582', 'SoIl32582224', datetime.datetime(2021, 1, 1, 9, 10), 1, 15, 12),
 (1, 'KiRu28935', 'KiRu28935224', datetime.datetime(2021, 1, 1, 9, 17), 1, 0, 46),
 (2, 'HaRu19752', 'HaRu19752224', datetime.datetime(2021, 1, 1, 9, 36), 1, 62, 40),
 (3, 'AkLu01628', 'AkLu01628224', datetime.datetime(2021, 1, 1, 9, 53), 1, 0, 1),
 (4, 'AkRu03357', 'AkRu03357224', datetime.datetime(2021, 1, 1, 10, 11), 1, 28, 14),
 (5, 'KiRu29315', 'KiRu29315224', datetime.datetime(2021, 1, 1, 10, 17), 1, 9, 40),
 (6, 'AkRu05234', 'AkRu05234224', datetime.datetime(2021, 1, 1, 10, 18), 1, 0, 30),
 (7, 'KiRu28520', 'KiRu28520224', datetime.datetime(2021, 1, 1, 10, 28), 1, 0, 34),
 (8, 'HaZa21742', 'HaZa21742224', datetime.datetime(2021, 1, 1, 10, 37), 1, 0, 6),
 (9, 'AmDa12214', 'AmDa12214224', datetime.datetime(2021, 1, 1, 10, 58), 1, 0, 36)]

1. How long did the survey take?

In [94]:
%%sql

SELECT
    DATEDIFF(max(time_of_record),min(time_of_record)) AS total_time_of_survey_in_days,
    DATEDIFF(,max(time_of_record),min(time_of_record)) AS total_time_of_survey_in_days
FROM
    visits

limit 10

 * mysql+pymysql://root:***@localhost:3306/md_water_services
1 rows affected.


[(924,)]

2. What is the average total queue time for water?


In [108]:
%%sql

SELECT
    AVG(NULLIF(time_in_queue, 0))
FROM
    visits
LIMIT
    50

 * mysql+pymysql://root:***@localhost:3306/md_water_services
1 rows affected.


[(Decimal('123.2574'),)]

3. What is the average queue time on different days?


In [115]:
%%sql

SELECT
    DAYNAME (time_of_record) AS Day_names,
    ROUND(AVG(NULLIF(time_in_queue, 0))) AS avg_time_in_queue
FROM
    visits
GROUP BY
    1
ORDER BY
    2 DESC

 * mysql+pymysql://root:***@localhost:3306/md_water_services
7 rows affected.


[('Saturday', Decimal('246')),
 ('Monday', Decimal('137')),
 ('Friday', Decimal('120')),
 ('Tuesday', Decimal('108')),
 ('Thursday', Decimal('105')),
 ('Wednesday', Decimal('97')),
 ('Sunday', Decimal('82'))]

4. How can we communicate this information efficiently?


In [123]:
%%sql

SELECT
    --    HOUR(time_of_record) AS hour_of_day_hour,
    TIME_FORMAT (TIME(time_of_record), '%H:00') AS hour_of_day,
    ROUND(AVG(NULLIF(time_in_queue, 0))) AS avg_time_in_queue
FROM
    visits
GROUP BY
    1
ORDER BY
    1

 * mysql+pymysql://root:***@localhost:3306/md_water_services
14 rows affected.


[('06:00', Decimal('149')),
 ('07:00', Decimal('149')),
 ('08:00', Decimal('149')),
 ('09:00', Decimal('118')),
 ('10:00', Decimal('114')),
 ('11:00', Decimal('111')),
 ('12:00', Decimal('112')),
 ('13:00', Decimal('115')),
 ('14:00', Decimal('114')),
 ('15:00', Decimal('114')),
 ('16:00', Decimal('114')),
 ('17:00', Decimal('149')),
 ('18:00', Decimal('147')),
 ('19:00', Decimal('168'))]

In [145]:
%%sql

SELECT
    TIME_FORMAT (TIME(time_of_record), '%H:00') AS hour_of_day,
    -- Sunday
    ROUND(
        AVG(
            CASE
                WHEN DAYNAME (time_of_record) = 'Sunday' THEN time_in_queue
                ELSE NULL
            END
        )
    ) AS Sunday,
    -- Monday
    ROUND(
        AVG(
            CASE
                WHEN DAYNAME (time_of_record) = 'Monday' THEN time_in_queue
                ELSE NULL
            END
        )
    ) AS Monday,
    -- Tuesday
    ROUND(
        AVG(
            CASE
                WHEN DAYNAME (time_of_record) = 'Tuesday' THEN time_in_queue
                ELSE NULL
            END
        )
    ) AS Tuesday,
    -- Wednesday
    ROUND(
        AVG(
            CASE
                WHEN DAYNAME (time_of_record) = 'Wednesday' THEN time_in_queue
                ELSE NULL
            END
        )
    ) AS Wednesday,
    -- Thursday
    ROUND(
        AVG(
            CASE
                WHEN DAYNAME (time_of_record) = 'Thursday' THEN time_in_queue
                ELSE NULL
            END
        )
    ) AS Thursday,
    -- Friday
    ROUND(
        AVG(
            CASE
                WHEN DAYNAME (time_of_record) = 'Friday' THEN time_in_queue
                ELSE NULL
            END
        )
    ) AS Friday,
    -- Saturday
    ROUND(
        AVG(
            CASE
                WHEN DAYNAME (time_of_record) = 'Saturday' THEN time_in_queue
                ELSE NULL
            END
        )
    ) AS Saturday
    
FROM
    visits
WHERE
    time_in_queue != 0 -- this exludes other sources with 0 queue times.
GROUP BY
    1
ORDER BY
    1

 * mysql+pymysql://root:***@localhost:3306/md_water_services
14 rows affected.


[('06:00', Decimal('79'), Decimal('190'), Decimal('134'), Decimal('112'), Decimal('134'), Decimal('153'), Decimal('247')),
 ('07:00', Decimal('82'), Decimal('186'), Decimal('128'), Decimal('111'), Decimal('139'), Decimal('156'), Decimal('247')),
 ('08:00', Decimal('86'), Decimal('183'), Decimal('130'), Decimal('119'), Decimal('129'), Decimal('153'), Decimal('247')),
 ('09:00', Decimal('84'), Decimal('127'), Decimal('105'), Decimal('94'), Decimal('99'), Decimal('107'), Decimal('252')),
 ('10:00', Decimal('83'), Decimal('119'), Decimal('99'), Decimal('89'), Decimal('95'), Decimal('112'), Decimal('259')),
 ('11:00', Decimal('78'), Decimal('115'), Decimal('102'), Decimal('86'), Decimal('99'), Decimal('104'), Decimal('236')),
 ('12:00', Decimal('78'), Decimal('115'), Decimal('97'), Decimal('88'), Decimal('96'), Decimal('109'), Decimal('239')),
 ('13:00', Decimal('81'), Decimal('122'), Decimal('97'), Decimal('98'), Decimal('101'), Decimal('115'), Decimal('242')),
 ('14:00', Decimal('83'), Decimal('127'), Decimal('104'), Decimal('92'), Decimal('96'), Decimal('110'), Decimal('244')),
 ('15:00', Decimal('83'), Decimal('126'), Decimal('104'), Decimal('88'), Decimal('92'), Decimal('110'), Decimal('248')),
 ('16:00', Decimal('83'), Decimal('127'), Decimal('99'), Decimal('90'), Decimal('99'), Decimal('109'), Decimal('251')),
 ('17:00', Decimal('79'), Decimal('181'), Decimal('135'), Decimal('121'), Decimal('129'), Decimal('151'), Decimal('251')),
 ('18:00', Decimal('80'), Decimal('174'), Decimal('122'), Decimal('113'), Decimal('132'), Decimal('158'), Decimal('240')),
 ('19:00', Decimal('127'), Decimal('159'), Decimal('145'), Decimal('176'), Decimal('137'), Decimal('103'), Decimal('282'))]

In [169]:
239 - 122 - 84

239

# 7-Reporting insights

**Insights**
1. Most water sources are rural.
2. 43% of our people are using shared taps. 2000 people often share one tap.
3. 31% of our population has water infrastructure in their homes, but within that group, 45% face non-functional systems due to issues with pipes,
pumps, and reservoirs.
4. 18% of our people are using wells of which, but within that, only 28% are clean..
5. Our citizens often face long wait times for water, averaging more than 120 minutes.
6. In terms of queues:
   - Queues are very long on Saturdays.
   - Queues are longer in the mornings and evenings.
   - Wednesdays and Sundays have the shortest queues.

**Start of our plan**

We have started thinking about a plan:
1. We want to focus our efforts on improving the water sources that affect the most people.
- Most people will benefit if we improve the shared taps first.
- Wells are a good source of water, but many are contaminated. Fixing this will benefit a lot of people.
- Fixing existing infrastructure will help many people. If they have running water again, they won't have to queue, thereby shorting queue times for
others. So we can solve two problems at once.
- Installing taps in homes will stretch our resources too thin, so for now, if the queue times are low, we won't improve that source.
2. Most water sources are in rural areas. We need to ensure our teams know this as this means they will have to make these repairs/upgrades in
rural areas where road conditions, supplies, and labour are harder challenges to overcome.

Practical solutions
1. If communities are using **rivers**, we can dispatch trucks to those regions to provide water temporarily in the short term, while we send out
crews to drill for wells, providing a more permanent solution.

2. If communities are using **wells**, we can install filters to purify the water. For wells with **biological** contamination, we can **install UV filters** that
kill microorganisms, and for \*polluted wells\*, we can install **reverse osmosis** filters. In the long term, we need to figure out why these sources
are polluted.

3. For **shared taps**, in the short term, we can send additional water tankers to the busiest taps, on the busiest days. We can use the queue time
pivot table we made to send tankers at the busiest times. Meanwhile, we can start the work on **installing extra taps** where they are needed.
According to UN standards, the maximum acceptable wait time for water is 30 minutes. With this in mind, our aim is to **install taps** to get
**queue times below 30 min**.

4. **Shared taps** with **short queue times (< 30 min)** represent a logistical challenge to further reduce waiting times. The most effective solution,
installing taps in homes, is resource-intensive and better suited as a long-term goal.

5. **Addressing broken infrastructure** offers a significant impact even with just a single intervention. It is expensive to fix, but so **many people**
can **benefit** from repairing one facility. For example, fixing a reservoir or pipe that multiple taps are connected to. We will have to find the
commonly affected areas though to see where the problem actually is.